nio_cli/commands/blockcheck.py

```diff
@@ -0,0 +1,241 @@
+                            key, block))
+
+                for prop, val in \
+                        spec_dict['nio/' + block]['properties'].items():
+                    if val['description'] == '':
+                        print(
+                            'Fill in the description for the '
+                            '"{}" property in the {} block'.format(prop, block)
+                        )
+        else:
+            print(
+                '\n**Run `nio buildspec {}` from the project directory '
+                'and re-run this check**\n'.format(self._block)
+            )
+            sys.exit()
+        print('')
+
+    def check_readme(self):
+
+        if os.path.exists('README.md'):
+            with open('README.md') as f:
+                lines = [l.rstrip() for l in f.readlines()]
+            block_indices = []
+            for block in self.blocks_in_spec:
+                if block not in lines:
+                    print('Add the {} block to the README')
+                block_indices.append(lines.index(block))
+            block_indices.sort()
+
+            for i in range(len(block_indices)):
+                for key in ['Properties', 'Inputs', 'Outputs',
+                            'Commands', 'Dependencies']:
+                    try:
+                        if key not in \
+                                lines[block_indices[i]:block_indices[i+1]]:
+                            print('Add "{}" to the {} block'.format(
+                                key, lines[block_indices[i]]))
+                    except IndexError:
+                        if key not in lines[block_indices[i]:]:
+                            print('Add "{}" to the {} block'.format(
+                                key, lines[block_indices[i]]))
+        else:
+            print(
+                'Run `nio buildreadme` '
+                'as long as the spec.json file is complete'
+            )
+        print('')
+
+    def check_release(self):
+
+        if os.path.exists('release.json'):
+            with open('release.json') as f:
+                release_dict = json.load(f)
+
+        for block in self.blocks_in_spec:
+            if 'nio/' + block not in release_dict:
+                print('Add {} block to release.json'.format(block))
+            for key in ['url', 'version', 'language']:
+                if key not in release_dict['nio/' + block] \
+                        or release_dict['nio/' + block][key] == '':
+                    print('Add a {} to the {} block'.format(key, block))
+            self.release_versions_dict[block] = \
+                release_dict['nio/' + block]['version']
+        print('')
+
+    def check_version(self):
+
+        for block in self.blocks_in_spec:
+            if self.release_versions_dict[block] != \
+                    self.spec_versions_dict[block]:
+                print(
+                    'Spec.json and release.json versions do not match for '
```

Comment:
These don't actually have to match. It's kind of confusing, but only the major and minor version number need to match. If the "patch" part of the version is off, that is ok.